bolt/internal/internal_test.go

```diff
@@ -133,8 +139,8 @@ func TestMarshalLayout(t *testing.T) {
 		t.Fatal(err)
 	} else if err := internal.UnmarshalLayout(buf, &vv); err != nil {
 		t.Fatal(err)
-	} else if !reflect.DeepEqual(layout, vv) {
-		t.Fatalf("source protobuf copy error: got %#v, expected %#v", vv, layout)
+	} else if !cmp.Equal(layout, vv) {
+		t.Fatal("source protobuf copy error: diff:\n", cmp.Diff(layout, vv))
```

Comment:
Not necessarily! We actually have to implement some logic to marshal and unmarshal (see `bolt/internal/internal.go`), and there's some generated code that depends on `bolt/internal/internal.proto`'s definitions. The desired behavior is tautological, but defects will violate that and cause this test to fail.

The replacement of `reflect.DeepEqual` with `cmp.Equal` is something I've been progressively doing as I work on tests since its equality is safer across Go versions than `DeepEqual`'s, and `cmp.Diff` produces the most amazing diffs evar.